cmd/argocd/commands/app.go

```diff
@@ -871,22 +800,107 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 	return command
 }
 
-func waitUntilOperationCompleted(appClient application.ApplicationServiceClient, appName string, timeout uint) (*argoappv1.Application, error) {
+func waitUntilOperationCompleted(appClient application.ApplicationServiceClient, appName string, timeout uint, watchSync, watchHealth, watchOperations bool) (*argoappv1.Application, error) {
```

Comment:
Lets improve the function name to be more generic, since it handles all three: sync, health, and operations. I propose: `waitOnApplicationStatus`?